.github/workflows/check_board.yml

```diff
@@ -0,0 +1,27 @@
+name: Check Board Headers
+
+on:
+  push:
+    paths:
+      - 'src/boards/include/boards/**'
```

Comment:
I guess there's _probably_ no reason that this shouldn't be shortened to just `'src/boards/**'`, given that nothing else lives in that directory?